src/components/CompareResults/ResultsTable.tsx

```diff
@@ -104,21 +104,78 @@ export default function ResultsTable() {
   const [searchTerm, setSearchTerm] = useState(initialSearchTerm);
   const [frameworkIdVal, setFrameworkIdVal] = useState(frameworkId);
   const [tableFilters, setTableFilters] = useState(
-    new Map() as Map<string, Set<string>>, // ColumnID -> Set<Values to remove>
+    new Map(), // ColumnID -> Set<Values to remove>
   );
+  const filterableKeys = ['status', 'platform', 'confidence'];
 
+  const getPossibleValues = (columnId: string): string[] | undefined => {
+    const config = cellsConfiguration.find(
+      (c) => c.key === columnId,
+    ) as CompareResultsTableConfig;
+    return config.possibleValues;
+  };
   const onClearFilter = (columnId: string) => {
+    const possibleValues = getPossibleValues(columnId);
     setTableFilters((oldFilters) => {
       const newFilters = new Map(oldFilters);
       newFilters.delete(columnId);
+      const values = possibleValues?.join(',') || '';
+      searchParams.set(columnId, values);
+      updateRawSearchParams(searchParams);
       return newFilters;
     });
   };
 
+  useEffect(() => {
+    const initialFilters = new Map() as Map<string, Set<string>>;
+    cellsConfiguration.forEach(({ key, possibleValues }) => {
+      if (!filterableKeys.includes(key) || !possibleValues) return;
+
+      const paramValue = searchParams.get(key);
+      const checkedValues = paramValue ? paramValue.split(',') : [];
+      const uncheckedValues = possibleValues?.filter(
+        (value: string) => !checkedValues.includes(value),
+      );
+      // when all filters are unchecked, persist state on reload
+      const values =
+        uncheckedValues.length === possibleValues.length
+          ? possibleValues
+          : uncheckedValues;
+
+      if (paramValue !== null) {
+        initialFilters.set(key, new Set(values));
+      } else {
+        searchParams.set(key, possibleValues.join(','));
+        initialFilters.set(key, new Set(possibleValues));
+      }
+    });
+
+    updateRawSearchParams(searchParams);
+    setTableFilters(initialFilters);
+  }, []);
```

Comment:
some suggestions to make this less complex:
1. use an effect with the "tableFilters" state to update the search params
2. do not set the state inside the effect, instead provide a default value when useState is called. It could be good to have functions to convert from the URL params to the filters and the other way around too.
3. try to not add anything to the URL if the default value is used
4. prefix the keys in the search param with some common prefix (maybe filter-XXX, eg filter-confidence)
5. I'm not sure yet about the value we use. Maybe it would be better to use some shorter keys instead of the full values, so that the displayed values can change and keep the URL working.

Happy to discuss this more if necessary!

Reply:
Thanks for these corrections. I was going with the idea that a user will want to know the filtered result they have based on the URL(mostly when shared) but I see how that complicates implementation. I'll work on these corrections.

Reply:
The idea was that in cellsConfiguratipn we'd have another property `urlValue`, as an array with the values we'd use for the URL for each of the values. The rationale is that it would be good to decorrelate the 2 uses:
1. one use is to display a value to the user => `possibleValues` (maybe it could be renamed to `displayedValues`? Not sure
2. one use is to add a value to the URL
3. and actually I can think of a 3rd use => the values we get from the server. Currently we use the same `possibleValues` array for this. But don't think about changing that in this PR, we can do that later.

So to decorrelate uses 1 and 2, we need 2 arrays for each of them.

Is that clearer?

Reply:
Thanks for the clarification, this makes sense now.